Mage.Sets/src/mage/cards/t/TurnTheTables.java

```diff
@@ -55,16 +55,15 @@ public TurnTheTablesEffect copy() {
     }
 
     @Override
-    protected PreventionEffectData preventDamageAction(GameEvent event, Ability source, Game game) {
-        PreventionEffectData preventionEffectData = super.preventDamageAction(event, source, game);
-        int damage = preventionEffectData.getPreventedDamage();
-        if (damage > 0) {
-            Permanent attackingCreature = game.getPermanent(getTargetPointer().getFirst(game, source));
-            if (attackingCreature != null) {
-                attackingCreature.damage(damage, source.getSourceId(), source, game, false, true);
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        DamageEvent damageEvent = (DamageEvent) event;
+        if (game.getPlayer(source.getControllerId() != null && game.getPermanent(source.getFirstTarget()) != null) {
```

Comment:
```suggestion
        if (game.getPlayer(source.getControllerId()) != null && game.getPermanent(source.getFirstTarget()) != null) {
```
You also missed as closing `)` for `game.getPlayer`.